resources/metadata/mavenInstall.yaml

```diff
@@ -0,0 +1,30 @@
+metadata:
+  name: mavenInstall
+  description: This step will install the maven project into the local maven repository.
+  longDescription: |
+    This step will install the maven project into the local maven repository.
+    It will also prepare jacoco to record the code coverage and
+    supports ci friendly versioning by flattening the pom before installing.
+spec:
```

Comment:
as @OliverNocon mentioned: I would also expect settingsHandling as an option. If you pass it to `maven.Execute` as config it will be evaluated

Reply:
yepp, if we want to expose it here (which I guess makes a lot of sense) then we should add respective step parameters ...

Reply:
settings handling? ah, I implemented it still with my groovy mindset that if I configure it for mavenExecute it will be used in here. But that's false I guess. 
However, then the same applies as for the Docker image, as I do not want to specify the settings handling for every step where maven is used but just once centrally.

Reply:
Any ideas?

Reply:
for the step parameters you can choose a scope.
If the scope is `GENERAL` the parameter can also be set in the general section of the configuration.

Ideally, in addition to the parameter name use an `alias` name which prevents possible confusion inside the general section.

Reply:
Will add this general thing in another pull request